python/doc/source/development.rst

```diff
@@ -118,7 +118,9 @@ dependencies will be automatically built by Arrow's third-party toolchain.
    $ sudo apt-get install libjemalloc-dev libboost-dev \
                           libboost-filesystem-dev \
                           libboost-system-dev \
-                          libboost-regex-dev
+                          libboost-regex-dev \
+                          flex \
```

Comment:
Preferably we list Thrift here instead of flex/bison. Is that available on all Debian-like systems?

Reply:
This gets built/statically linked by parquet-cpp -- we probably want to use the vendored Thrift to avoid version discrepancies depending on the distribution (since Thrift recently broke some shared_ptr APIs from 0.10 to 0.11)

Reply:
So I do not know what should be used here exactly. I was just compiling it and this is what cmake said it is missing. First flex and then bison. If thrift provides both (or an alternative) and cmake picks it up, this is great. But it compiled with flex and bison for me well.

Reply:
Is this the package we would need: https://packages.debian.org/stretch/thrift-compiler

If it is, I think this works across all Debian-like systems.

Reply:
We use the Thrift C++ libraries so the compiler alone is not suffiicient

Reply:
Why I didn't have to install those libraries? I had to install flex and bison.

Reply:
(I do not have anything from thrift installed on the system either.)

Reply:
If thrift is not installed on your system, we will build it during the Arrow build (but to build Thrift, we need flex&bison).